stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/imagepreview/ImagePreviewActivity.kt

```diff
@@ -141,35 +265,245 @@ public class ImagePreviewActivity : AppCompatActivity() {
+            )
+        }
+    }
+
+    /**
+     * Consumes the action user selected to perform for the current image.
+     *
+     * @param imagePreviewAction The action the user selected.
+     * @param currentPage The index of the current image.
+     */
+    private fun handleImageAction(
+        imagePreviewAction: ImagePreviewAction,
+        currentPage: Int,
+    ) {
+        val message = imagePreviewAction.message
+
+        when (imagePreviewAction) {
+            is ShowInChat -> {
+                handleResult(
+                    ImagePreviewResult(
+                        messageId = message.id,
+                        resultType = ImagePreviewResultType.SHOW_IN_CHAT
+                    )
+                )
+            }
+            is Reply -> {
+                handleResult(ImagePreviewResult(messageId = message.id, resultType = ImagePreviewResultType.QUOTE))
+            }
+            is Delete -> imagePreviewViewModel.deleteCurrentImage(message.attachments[currentPage])
+            is SaveImage -> {
+                ChatDomain
+                    .instance()
+                    .downloadAttachment(message.attachments[currentPage])
+                    .enqueue()
+            }
+        }
+    }
+
+    /**
+     * Prepares and sets the result of this Activity and propagates it back to the user.
+     *
+     * @param result The chosen action result.
+     */
+    private fun handleResult(result: ImagePreviewResult) {
+        val data = Intent().apply {
+            putExtra(KEY_IMAGE_PREVIEW_RESULT, result)
         }
+        setResult(RESULT_OK, data)
+        finish()
     }
 
+    /**
+     * Renders a horizontal pager that shows images and allows the user to swipe them and zoom in.
+     *
+     * @param pagerState The state of the pager, that represents the page count and the current page.
+     * @param attachments The attachments to show.
+     */
     @Composable
-    private fun ImagePreviewContent(pagerState: PagerState, message: Message) {
+    private fun ImagePreviewContent(pagerState: PagerState, attachments: List<Attachment>) {
         HorizontalPager(modifier = Modifier.background(ChatTheme.colors.appBackground), state = pagerState) { page ->
 
-            if (message.attachments.isNotEmpty()) {
-                val painter = rememberImagePainter(data = message.attachments[page].imagePreviewUrl)
+            if (attachments.isNotEmpty()) {
+                val painter = rememberImagePainter(data = attachments[page].imagePreviewUrl)
+
+                var currentScale by remember { mutableStateOf(1f) }
+
+                val transformableState = rememberTransformableState { zoomChange, _, _ ->
+                    val newScale = (currentScale * zoomChange)
+                        .coerceAtLeast(1f)
+                        .coerceAtMost(3f)
+
+                    currentScale = newScale
+                }
+
+                val scale by animateFloatAsState(targetValue = currentScale)
```

Comment:
This scaling works for an initial implementation, but we should revisit this and add nicer scale/pan control later on.

Reply:
In this case it's kind of the default - zoom in 2x, then zoom in to 3x, then zoom out on taps. 

Or you mean the animation?

Reply:
I mean we're just scaling into the middle of the image regardless of where the taps are happening, and then we can't pan around on the zoomed image

Reply:
Oh okay! That makes sense - I tried to keep it as close to the UI comp implementation which is fairly basic.

In Compose, we can easily add rotations, panning etc if needed.